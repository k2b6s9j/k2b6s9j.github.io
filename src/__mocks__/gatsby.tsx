import * as React from "react";
const gatsby = jest.requireActual("gatsby");
import { graphql } from "graphql";

type MockedGraphQLReturn = Promise<any>;

const mockedGraphQLQuery = (query: string): MockedGraphQLReturn =>
  graphql(undefined, query);

interface StaticQueryHOCProps {
  query: MockedGraphQLReturn;
  render?: (data: object) => React.ReactElement;
  children?: (data: object) => React.ReactElement;
}

const mockedStaticQueryHOC = (
  props: StaticQueryHOCProps
): React.ReactElement => {
  if (!props.children || !props.render) {
    throw new Error("Render prop must be provided!");
  }

  return (
    (props.children && props.children({})) ||
    (props.render && props.render({})) || <div>Error!</div>
  );
};

const mockedStaticQueryHook = (data: MockedGraphQLReturn): object => ({});

module.exports = {
  ...gatsby,
  graphql: jest.fn(mockedGraphQLQuery),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      to,
      ...rest
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(mockedStaticQueryHOC),
  useStaticQuery: jest.fn(mockedStaticQueryHook)
};
