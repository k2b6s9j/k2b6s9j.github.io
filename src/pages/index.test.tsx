import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./index";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    file: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    image: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    backdrop: {
      childImageSharp: {
        fluid: {
          base64: "",
          aspectRatio: "",
          src: "",
          srcSet: "",
          srcWebp: "",
          srcSetWebp: "",
          sizes: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: "",
        description: "",
        siteUrl: "",
        twitterUsername: "",
        instagramUsername: "",
        linkedinUsername: "",
        githubUsername: ""
      }
    }
  }));
});

describe("Index Page", (): void => {
  it("renders correctly", (): void => {
    const tree = renderer
      .create(
        <Page
          path="/"
          data={{
            site: {
              siteMetadata: {
                title: "",
                description: "",
                siteUrl: "",
                twitterUsername: "",
                instagramUsername: "",
                linkedinUsername: "",
                githubUsername: ""
              }
            },
            metadataImage: {
              childImageSharp: {
                fixed: {
                  base64: "",
                  width: 100,
                  height: 100,
                  src: "",
                  srcSet: ""
                }
              }
            }
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
