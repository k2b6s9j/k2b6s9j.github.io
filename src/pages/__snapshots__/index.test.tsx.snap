// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
Array [
  <head>
    Object {}
    Object {
      "lang": "en",
    }
    <meta
      content="Kepler Sticka-Jones"
      data-react-helmet={true}
      name="author"
    />
    <meta
      content=""
      data-react-helmet={true}
      name="description"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:site_name"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:title"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:description"
    />
    <meta
      content="summary"
      data-react-helmet={true}
      name="twitter:card"
    />
    <meta
      content="@test"
      data-react-helmet={true}
      name="twitter:site"
    />
    <meta
      content="@test"
      data-react-helmet={true}
      name="twitter:creator"
    />
    <title
      data-react-helmet={true}
    >
      
    </title>
  </head>

<script
    type="application/ld+json"
  >
    Object {
    "@context": "https://schema.org",
    "@type": "WebSite",
    "about": Object {
      "@type": "Person",
      "description": "",
      "image": "",
      "name": "Kepler Sticka-Jones",
      "sameAs": Array [
        "https://twitter.com/test",
        "https://github.com/test",
      ],
      "url": "",
    },
    "description": "",
    "name": "Kepler Sticka-Jones",
    "url": "",
  }
  </script>,
  .emotion-12 {
  min-height: 100vh;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 3rem 1.5rem;
}

.emotion-10 {
  text-align: center !important;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 0 auto;
  position: relative;
  width: auto;
}

@media screen and (min-width:1024px) {
  .emotion-10 {
    max-width: 960px;
  }
}

@media screen and (min-width:1216px) {
  .emotion-10 {
    max-width: 1152px;
  }
}

.emotion-9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -0.75rem;
  margin-right: -0.75rem;
  margin-top: -0.75rem;
}

.emotion-9:last-child {
  margin-bottom: -0.75rem;
}

.emotion-8 {
  margin-left: auto;
  margin-right: auto;
  display: block;
  padding: 0.75rem;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 66.6666%;
}

.emotion-0 {
  margin-bottom: 10px;
  border-radius: 140px;
}

.emotion-1 {
  font-size: 2.5rem;
  font-weight: 600;
  line-height: 1.125;
}

.emotion-7 {
  margin-left: auto;
  margin-right: auto;
}

.emotion-2 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 2.25em;
  line-height: 1.5;
  position: relative;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-bottom: calc(0.375em - 1px);
  padding-left: 0.75em;
  padding-right: 0.75em;
  padding-top: calc(0.375em - 1px);
  text-align: center;
  white-space: nowrap;
  border-color: transparent;
  font-size: 1.5rem;
}

.emotion-2:hover {
  background-color: #f9f9f9;
}

.emotion-2:active {
  background-color: #f2f2f2;
}

@media (prefers-color-scheme:dark) {
  .emotion-2:hover {
    background-color: #606060;
  }

  .emotion-2:active {
    background-color: #d0d0d0;
  }
}

<section
    className="emotion-12"
  >
    <div
      className="emotion-11"
    >
      <div
        className="emotion-10"
      >
        <div
          className="emotion-9"
        >
          <div
            className="emotion-8"
          >
            <div
              aria-label="Avatar"
              role="img"
            >
              <div
                className="emotion-0 gatsby-image-wrapper"
                style={
                  Object {
                    "display": "inline-block",
                    "height": 100,
                    "overflow": "hidden",
                    "position": "relative",
                    "width": 100,
                  }
                }
              >
                
                <picture>
                  <source
                    srcSet=""
                  />
                  <img
                    alt="Avatar"
                    height={100}
                    loading="lazy"
                    onLoad={[Function]}
                    src=""
                    srcSet=""
                    style={
                      Object {
                        "height": "100%",
                        "left": 0,
                        "objectFit": "cover",
                        "objectPosition": "center",
                        "opacity": 0,
                        "position": "absolute",
                        "top": 0,
                        "transition": "opacity 500ms",
                        "width": "100%",
                      }
                    }
                    width={100}
                  />
                </picture>
                <noscript
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<picture><source srcset=\\"\\" /><img loading=\\"lazy\\" width=\\"100\\" height=\\"100\\" src=\\"\\" alt=\\"Avatar\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                    }
                  }
                />
              </div>
            </div>
            <h1
              className="emotion-1"
            >
              Kepler Sticka-Jones
            </h1>
            <br />
            <div
              className="emotion-7"
            >
              <a
                className="emotion-2"
                href="https://github.com/test"
              >
                Code
              </a>
              <a
                className="emotion-2"
                href="https://twitter.com/test"
              >
                Tweets
              </a>
              <a
                className="emotion-2"
                href="/blog"
              >
                Blog
              </a>
              <a
                className="emotion-2"
                href="/projects"
              >
                Projects
              </a>
              <a
                className="emotion-2"
                href="/about"
              >
                About
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
